solr-ingest/src/test/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilterTest.java

```diff
@@ -0,0 +1,152 @@
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.search.solr.model.IndexDocumentBean;
+import edu.unc.lib.dl.search.solr.util.FacetConstants;
+
+/**
+ * @author harring
+ */
+public class SetAccessStatusFilterTest {
+
+    private static final String PID_STRING = "uuid:07d9594f-310d-4095-ab67-79a1056e7430";
+
+    @Mock
+    private DocumentIndexingPackage dip;
+    @Mock
+    private IndexDocumentBean idb;
+    @Mock
+    private InheritedAclFactory inheritedAclFactory;
+    @Mock
+    private ObjectAclFactory objAclFactory;
+    @Mock
+    private ContentObject contentObj;
+    @Mock
+    private Date date;
+
+    @Mock
+    private PID pid;
+
+    @Captor
+    private ArgumentCaptor<List<String>> listCaptor;
+
+    private Map<String, Set<String>> principalRoles;
+
+    private SetAccessStatusFilter filter;
+
+    @Before
+    public void setup() throws Exception {
+        initMocks(this);
+
+        when(pid.toString()).thenReturn(PID_STRING);
+
+        when(dip.getDocument()).thenReturn(idb);
+        when(dip.getPid()).thenReturn(pid);
+        when(dip.getContentObject()).thenReturn(contentObj);
+        when(contentObj.getPid()).thenReturn(pid);
+
+        when(inheritedAclFactory.isMarkedForDeletion(any(PID.class))).thenReturn(true);
+
+        principalRoles = new HashMap<>();
+        principalRoles.put(AccessPrincipalConstants.PUBLIC_PRINC, new HashSet<String>());
+        when(inheritedAclFactory.getPrincipalRoles(any(PID.class))).thenReturn(principalRoles);
+        // default for test suite is staff-only access; changes made within test cases as required
+        when(inheritedAclFactory.getPatronAccess(any(PID.class))).thenReturn(PatronAccess.none);
+        when(objAclFactory.getPatronAccess(any(PID.class))).thenReturn(PatronAccess.none);
+
+        filter = new SetAccessStatusFilter();
+        filter.setObjectAclFactory(objAclFactory);
+        filter.setInheritedAclFactory(inheritedAclFactory);
+    }
+
+    @Test
+    public void testIsMarkedForDeletion() throws Exception {
+
+        filter.filter(dip);
+
+        verify(idb).setStatus(listCaptor.capture());
+        assertTrue(listCaptor.getValue().contains(FacetConstants.MARKED_FOR_DELETION));
+    }
+
+    @Test
+    public void testIsEmbargoed() throws Exception {
+
+        when(inheritedAclFactory.getEmbargoUntil(any(PID.class))).thenReturn(date);
+
+        filter.filter(dip);
+
+        verify(idb).setStatus(listCaptor.capture());
+        assertTrue(listCaptor.getValue().contains(FacetConstants.EMBARGOED));
+    }
+
+    @Test
+    public void testHasStaffOnlyAccess() throws Exception {
+
+        filter.filter(dip);
+
+        verify(idb).setStatus(listCaptor.capture());
+        assertTrue(listCaptor.getValue().contains(FacetConstants.STAFF_ONLY_ACCESS));
+    }
+
+    @Test
+    public void testHasPublicAccess() throws Exception {
+        when(inheritedAclFactory.getPatronAccess(any(PID.class))).thenReturn(PatronAccess.everyone);
```

Comment:
I don't think this is necessary, and isn't a valid result from getPatronAccess for a folder/work/fileObj.  If you have PatronAccess.parent as the default for the tests that'll take care of this case.  I'd also suggest moving 
`principalRoles.put(AccessPrincipalConstants.PUBLIC_PRINC, new HashSet<String>());`
from the init method down to this test so that you can be sure it's not affecting other unrelated tests.